src/docs/Core-2.0/Categories/CategoriesHelper.md

```diff
@@ -0,0 +1,138 @@
+Categories helper services
+==========================
+
+For some uncommon tasks you may need advanced functionality that is not provided by the category API services. In such situations you can also use additional helper services the core provides, too.
+
+Processing helper
+-----------------
+
+classname: \Zikula\CategoriesModule\Helper\CategoryProcessingHelper
+
+service id = "zikula_categories_module.category_processing_helper"
+
+Contains methods used for processing categories for input or output.
+
+The class makes the following methods available:
+
+    - validateCategoryData($data)
+    - processCategoryName($name)
+    - processCategoryParent($parentId)
+    - processCategoryDisplayName($displayName, $name)
+    - processCategoryPath($parentPath, $categoryName)
+    - processCategoryIPath($parentIpath, $categoryId)
+    - processCategoryAttributes($category, $attribNames, $attribValues)
+    - mayCategoryBeDeletedOrMoved($category)
+
+Sorting helper
+--------------
+
+classname: \Zikula\CategoriesModule\Helper\CategorySortingHelper
+
+service id = "zikula_categories_module.category_sorting_helper"
+
+Contains methods used for performing or amending the sorting of categories.
+
+The class makes the following methods available:
+
+    - cmpName($catA, $catB)
+    - cmpDesc($catA, $catB)
+    - sortByLocale(&$cats, $func = 'cmpName')
+    - resequence($cats, $step = 1)
+    - _tree_insert(&$tree, $entry, $currentpath = null)
```

Comment:
also, rename methods in new Apis to be descriptive - e.g. cmpName() to compareName() etc.